pkg/cri/server/server.go

```diff
@@ -199,6 +205,39 @@ func (s *server) createGrpcServer() error {
 	return nil
 }
 
+// Chmod changes the permissions of the server's socket.
+func (s *server) Chmod(mode os.FileMode) error {
+	err := os.Chmod(s.options.Socket, mode)
+	if err != nil {
+		s.Error("failed to change permissions of socket %q to %v: %v",
+			s.options.Socket, mode, err)
+	} else {
+		s.Info("changed permissions of socket %q to %v", s.options.Socket, mode)
+	}
+	return err
+}
+
+// Chown changes ownership of the server's socket.
+func (s *server) Chown(uid, gid int) error {
+	usr := strconv.FormatInt(int64(uid), 10)
+	if u, err := user.LookupId(usr); u != nil && err == nil {
+		usr = u.Name
+	}
+	grp := strconv.FormatInt(int64(gid), 10)
+	if g, err := user.LookupGroupId(grp); g != nil && err == nil {
+		grp = g.Name
+	}
```

Comment:
nit: These could be moved down to the else branch where they are actually used. Would make the code easier to read. Prolly it's just me plus the late evening but it took a while for me to grasp that these are actually only for human-readability

Reply:
It's also probably partly the weird golang API: you have to pass integer arguments as strings. Anyway, the resolved usr/grp was supposed to be used in both the success and the failure branches, that's why the lookups are in the beginning. Updated the code accordingly.